tests/unit/test_utils.py

```diff
@@ -332,26 +425,62 @@ def confirm_files(self):
                 "mode: %s, expected mode: %s" % (mode, expected_mode)
             )
 
-    def test_unpack_tgz(self, data):
+    def test_unpack_tar(self):
         """
-        Test unpacking a *.tgz, and setting execute permissions
+        Test unpacking a *.tar, and setting execute permissions
         """
-        test_file = data.packages.join("test_tar.tgz")
+        test_file = self.create_tar_files()
```

Comment:
I would leave this test alone (along with `test_unpack_zip()` and the helper functions associated with those tests) because it is introducing a lot of unnecessary complexity into the PR and making it harder to review. If you'd like to refactor existing tests, it can be proposed in a separate PR. It's better to keep the PR smaller and more focused on the issue at hand.